src/TestCases/TestingTypeMaster/TestingTypeMasterComponent.js

```diff
@@ -0,0 +1,239 @@
+    type: "",
+    data: "",
+    open: false,
+  });
+
+  // Function to handle search button click
+  const handleSearch = () => {
+    const filteredList = customSearchHandler(
+      testingTypeMasetrList,
+      searchValue
+    );
+    setFilterTestingTypeMasterList(filteredList);
+  };
+
+  // Function to handle reset button click
+  const handleReset = () => {
+    setSearchValue("");
+    setFilterTestingTypeMasterList(testingTypeMasetrList);
+  };
+
+  // Update the useEffect to update the filtered list when testingTypeMasetrList changes
+  useEffect(() => {
+    setFilterTestingTypeMasterList(testingTypeMasetrList);
+  }, [testingTypeMasetrList]);
+
+  // Function to handle search onchange
+  useEffect(() => {
+    handleSearch();
+  }, [searchValue]);
+
+  const columns = [
+    {
+      name: "Sr. No.",
+      selector: (row, index) => index + 1,
+      sortable: false,
+      width: "70px",
+    },
+    {
+      name: "Action",
+      selector: (row) => (
+        <i
+          className="icofont-edit text-primary cp"
+          onClick={() =>
+            setAddEditTestingTypeModal({
+              type: "EDIT",
+              data: row,
+              open: true,
+            })
+          }
+        />
+      ),
+      sortable: false,
+      width: "70px",
+    },
+
+    {
+      name: "Status",
+      selector: (row) => (row.is_active === 1 ? "Active" : "Deactive"),
+      sortable: false,
+      width: "100px",
+    },
+
+    {
+      name: "Testing Type Title",
+      selector: (row) => row.type_name,
+      sortable: false,
+      width: "200px",
+    },
+
+    {
+      name: "Created At",
+      selector: (row) => row.created_at,
+      sortable: false,
+      width: "175px",
+    },
+
+    {
+      name: "Created By",
+      selector: (row) => row.created_by,
+      sortable: false,
+      width: "175px",
+    },
+    {
+      name: "Updated At",
+      selector: (row) => row.updated_at,
+      sortable: false,
+      width: "175px",
+    },
+
+    {
+      name: "Updated By",
+      selector: (row) => row.updated_by,
+      sortable: false,
+      width: "175px",
+    },
+  ];
+
+  const transformDataForExport = (data) => {
+    return data?.map((row, index) => ({
+      "Sr No.": index + 1,
+      "Testing Type Title": row?.type_name || "--",
+      "Created At": row?.created_at || "--",
+      "Created By": row?.created_by || "--",
+      "Updated At": row?.updated_at || "--",
+      "Updated By": row?.updated_by || "--",
+    }));
+  };
+
+  useEffect(() => {
+    dispatch(getTestingTypeMasterListThunk());
+  }, []);
+
+  return (
+    <div className="container-xxl">
+      <PageHeader
+        headerTitle="Testing Type Master"
+        renderRight={() => {
+          return (
+            <div className="col-md-6 d-flex justify-content-end">
+              <div className="justify-content-between">
+                <button
+                  className="btn btn-primary text-white"
+                  style={{
```

Comment:
1.Inline css is strictly not allowed in project.
Its only allowed when there is dynamic styling or considitional based styling.

Reply:
remove inline style and applying dynamic style